src/components/flat-table/flat-table.pw.tsx

```diff
@@ -0,0 +1,3619 @@
+    await checkFocus(flatTableBodyRowByPosition(page, 3));
+  });
+
+  test(`should render with clickable rows accessible using down arrow keys when focusRedesignOptOut is false`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<FlatTableAllSubrowSelectableComponent />);
+
+    await page.keyboard.press("Tab");
+    for (let i = 0; i < 5; i++) {
+      await page.keyboard.press("Tab");
+    }
+
+    await checkNewFocusStyling(flatTableBodyRowByPosition(page, 0));
+    await page.keyboard.press("Tab");
+    await expect(
+      flatTableBodyRowByPosition(page, 0).locator("input")
+    ).toBeFocused();
+    await page.keyboard.press("ArrowDown");
+    await checkNewFocusStyling(flatTableBodyRowByPosition(page, 1));
+    await page.keyboard.press("ArrowDown");
+    await checkNewFocusStyling(flatTableBodyRowByPosition(page, 2));
+    await page.keyboard.press("ArrowDown");
+    await checkNewFocusStyling(flatTableBodyRowByPosition(page, 3));
+    await page.keyboard.press("ArrowDown");
+    await checkNewFocusStyling(flatTableBodyRowByPosition(page, 3));
+  });
+
+  test(`should render with clickable rows accessible using up arrow keys when focusRedesignOptOut is true`, async ({
+    mount,
+    page,
+  }) => {
+    await mount<HooksConfig>(<FlatTableAllSubrowSelectableComponent />, {
+      hooksConfig: { focusRedesignOptOut: true },
+    });
+
+    await page.keyboard.press("Tab");
+    const bodyRowByPos3 = flatTableBodyRowByPosition(page, 3)
+      .locator("input")
+      .nth(0);
+    await bodyRowByPos3.focus();
+    await expect(bodyRowByPos3).toBeFocused();
+    await page.keyboard.press("ArrowUp");
+    await checkFocus(flatTableBodyRowByPosition(page, 2));
+    await page.keyboard.press("ArrowUp");
+    await checkFocus(flatTableBodyRowByPosition(page, 1));
+    await page.keyboard.press("ArrowUp");
+    await checkFocus(flatTableBodyRowByPosition(page, 0));
+    await page.keyboard.press("ArrowUp");
+    await checkFocus(flatTableBodyRowByPosition(page, 0));
+  });
+
+  test(`should render with clickable rows accessible using up arrow keys when focusRedesignOptOut is false`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<FlatTableAllSubrowSelectableComponent />);
+
+    await page.keyboard.press("Tab");
+    const bodyRowByPos3 = flatTableBodyRowByPosition(page, 3)
+      .locator("input")
+      .nth(0);
+    await bodyRowByPos3.focus();
+    await expect(bodyRowByPos3).toBeFocused();
+    await page.keyboard.press("ArrowUp");
+    await checkNewFocusStyling(flatTableBodyRowByPosition(page, 2));
+    await page.keyboard.press("ArrowUp");
+    await checkNewFocusStyling(flatTableBodyRowByPosition(page, 1));
+    await page.keyboard.press("ArrowUp");
+    await checkNewFocusStyling(flatTableBodyRowByPosition(page, 0));
+    await page.keyboard.press("ArrowUp");
+    await checkNewFocusStyling(flatTableBodyRowByPosition(page, 0));
+  });
+
+  ["ArrowLeft", "ArrowRight"].forEach((arrow) => {
```

Comment:
suggestion (non-blocking): store in a variable, less of an issue here as the tests using it are right next to each other and unlikely we update the values etc